File: python/src/coding_interview/application/repository/market_price_repository.py
from __future__ import annotations

from decimal import Decimal
from typing import Protocol

from coding_interview.domain.stock_symbol import StockSymbol


class MarketPriceRepository(Protocol):
    """市場価格リポジトリ。"""

    def all(self) -> dict[StockSymbol, Decimal]: ...
    def update(self, prices: dict[StockSymbol, Decimal]) -> None: ...

File: php/src/CodingInterview/Application/Repository/MarketPriceRepository.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Application\Repository;

use Folio\CodingInterview\Domain\BigDecimal;

/**
 * 市場価格リポジトリ。
 */
interface MarketPriceRepository
{
    /** @return array<string,BigDecimal> keyed by StockSymbol->value */
    public function all(): array;

    /** @param array<string,BigDecimal> $prices */
    public function update(array $prices): void;
}
